mitodl/open-discussions
# Handle Forbidden exception

#### What are the relevant tickets?
Fixes #319 

#### What's this PR do?
Converts PRAW exceptions to DRF exceptions so that we get the 403 or 404 responses. This PR will only handle this for the cases where we are relying on reddit to do these permission checks.

#### How should this be manually tested?
Open up the network tab in your browser and make sure these return these results:
 - Go to `/channel/missing` and verify that the API errors are all 404s
 - For a channel you are subscribed to go to `/channel/channel_name/missing/` and verify the API errors are 404s
 - Create a private channel your user is not subscribed to. Go to `/channel/<channel_name>/` and verify the API errors are all 403s

There are many other cases which may not be worth testing manually, but in general you should never see a 500 error from an API call.
